crates/symbolicator-service/src/services/download/sentry.rs

```diff
@@ -37,19 +37,41 @@ pub struct SearchArtifactResult {
     pub headers: BTreeMap<String, String>,
 }
 
+/// We would want to cache differently-shaped results in our lookup cache
+#[derive(Clone, Debug)]
+enum SearchResult {
+    JustId(Vec<JustIdResult>),
+    Artifact(Vec<ArtifactResult>),
+}
+
+trait IntoSearchResult: Sized {
+    fn into_search_result(vec: Vec<Self>) -> SearchResult;
+}
+
+impl IntoSearchResult for JustIdResult {
+    fn into_search_result(vec: Vec<Self>) -> SearchResult {
+        SearchResult::JustId(vec)
+    }
+}
+impl IntoSearchResult for ArtifactResult {
+    fn into_search_result(vec: Vec<Self>) -> SearchResult {
+        SearchResult::Artifact(vec)
+    }
+}
+
```

Comment:
You can us `From` for this.

```suggestion
impl From<Vec<JustIdResult>> for SearchResult {
    fn from(val: Vec<JustIdResult>) -> Self {
        SearchResult::JustId(val)
    }
}

impl From<Vec<ArtifactResult>> for SearchResult {
    fn from(val: Vec<ArtifactResult>) -> Self {
        SearchResult::Artifact(val)
    }
}
```